editor/editor_node.cpp

```diff
@@ -6980,8 +6980,8 @@ EditorNode::EditorNode() {
 
 	main_vbox = memnew(VBoxContainer);
 	gui_base->add_child(main_vbox);
-	main_vbox->set_anchors_and_offsets_preset(Control::PRESET_FULL_RECT, Control::PRESET_MODE_MINSIZE, 8);
-	main_vbox->add_theme_constant_override("separation", 8 * EDSCALE);
+	main_vbox->set_anchors_and_offsets_preset(Control::PRESET_FULL_RECT, Control::PRESET_MODE_MINSIZE, gui_base->get_theme_constant(SNAME("window_border_margin"), SNAME("Editor")));
+	main_vbox->add_theme_constant_override("separation", gui_base->get_theme_constant(SNAME("top_bar_separation"), SNAME("Editor")));
```

Comment:
You also need to set this when the theme is changed.

Reply:
Would you have an example? I notice most of the controls do not redefine those when they receive NOTIFICATION_THEME_CHANGED; they only remeasure their content.

Reply:
This is incorrect. Whenever you read from the theme you should do it every time any theme affecting your control changes. All built-in controls do it: simple ones just use them in `NOTIFICATION_DRAW` directly, and more complex ones apply what they need to their children (similar to what you are doing here).

In editor this is sometimes missed, but that's a bug, not the correct behavior. `EditorNode` that you're editing is a special case on top of that, because a) it is not a control node nor a window, so it doesn't receive `NOTIFICATION_THEME_CHANGED`; and b) it defines the base theme for the entire editor itself. So in case of `EditorNode` you want to add this line in the `EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED` handler, after the new theme is set. There are similar lines there already to give you a point of reference.

Reply:
Thanks for clarifying. I edited the part you mentioned. Can you have a look and confirm it is in the right place?